version: '3.9'

services:
  db:
    image: postgres:16.3-alpine3.20
    volumes:
      - postgres_data:/var/lib/postgresql/data
    environment:
      POSTGRES_DB: ejercicio_db
      POSTGRES_USER: postgres
      POSTGRES_PASSWORD: admin
    ports:
      - "5434:5432"
    healthcheck:
      test: ["CMD", "pg_isready", "-U", "postgres"]
      interval: 5s
      timeout: 5s
      retries: 5

  app:
    build: ./drf-api
    volumes:
      - ./drf-api:/app
    environment:
      - DATABASE_HOST=db
      - DATABASE_USER=postgres
      - DATABASE_PASSWORD=admin
      - DATABASE_NAME=ejercicio_db
    depends_on:
      db:
        condition: service_healthy
    ports:
      - "8000:8000"
    #command: ["python3", "manage.py", "runserver", "0.0.0.0:8000"]

  frontend:
    build: ./vue-ui
    ports:
      - "8080:8080"
    command: ["npm", "run", "dev"]
    volumes:
      - ./vue-ui:/app
      - /app/node_modules
    # environment:
    #   - VITE_BACKEND_URL=http://app:8000

volumes:
  postgres_data:
